### PR TITLE
kinetis_common: add a option to enable hardware CS for kinetis spi

### PR DESCRIPTION
**Pull request by jfischer-phytec-iot**
This PR adds a option to enable hardware CS for kinetis spi (PCS0 only) + some fixes.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@jfischer-phytec-iot I think I can live with that workaround, after all the discussions about th CS line :-). @gebart do you have time to test?

File: cpu/kinetis_common/spi.c [hunk 1]
```diff
@@ -22,6 +22,10 @@
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
+#ifndef KINETIS_SPI_USE_HW_CS
+#define KINETIS_SPI_USE_HW_CS   0
+#endif
+
```
Comment (PeterKietzmann) [inline cpu/kinetis_common/spi.c]: Wouldn't it make sense to move this switch to the periph_conf?

Comment (jfischer-phytec-iot) [inline cpu/kinetis_common/spi.c]: @PeterKietzmann  No, it is also necessary for the lines https://github.com/jfischer-phytec-iot/RIOT/commit/ba9db2d147ac99fe0efaa9e6d7d924708f393e52#diff-1cbad4f25cbb4ed1edc201509a26b427R346 and https://github.com/jfischer-phytec-iot/RIOT/commit/ba9db2d147ac99fe0efaa9e6d7d924708f393e52#diff-1cbad4f25cbb4ed1edc201509a26b427R764.

**Comment by PeterKietzmann** [comment]
Still I wonder why you cannot move it to the periph_conf.. I have a phyWAVE-A-Eval-Board here, is it possible to test SPI with this board? And by the way: what happened to "RIOT/boards/pba-d-01-kw2x/*" ?

**Comment by jfischer-phytec-iot** [comment]
>Still I wonder why you cannot move it to the periph_conf

It allows backward compatibility, if ```#define KINETIS_SPI_USE_HW_CS 1``` does not exist in periph_conf.h.

>I have a phyWAVE-A-Eval-Board here, is it possible to test SPI with this board?

Sure, just rebase #2059 on this PR, btw do you have time to review #2059 :smile: ?

>And by the way: what happened to "RIOT/boards/pba-d-01-kw2x/*" ?

:question: need more input ...

**Comment by PeterKietzmann** [comment]
@jfischer-phytec-iot I will test this PR first. Your code changes look reasonable to me. Regarding a review of #2059 I can't promise anything at the moment but I'll try to keep it in mind. 
The question about the pba-d-01 was just a personal confusion. For some reasons I thought I was already in the RIOT master...

**Comment by PeterKietzmann** [comment]
@jfischer-phytec-iot I've got a phyWAVE-A-Eval-Board (1st revision). Are the relevant SPI pins (including CS line) on the pinout? If so, on which pins are they?

**Comment by jfischer-phytec-iot** [comment]
@PeterKietzmann thx

**Comment by PeterKietzmann** [comment]
You're welcome! Will you also answer my question :-) ?

**Comment by gebart** [comment]
I am currently swamped at work and will not be able to test any of this in the near future, sorry.

**Comment by jfischer-phytec-iot** [comment]
@PeterKietzmann :smiley: look at your mailbox

**Comment by PeterKietzmann** [comment]
@jfischer-phytec-iot I tend to ACK after you've squashed your commits and seeing travis happy. Here is my result for the `SPI_0` bus (one can also see the benifit of hardware over software CS line):
bildschirmfoto vom 2015-04-16 11-31-41
Anyone NACKing this? I'm not too familiar with the kinetis series.

**Comment by PeterKietzmann** [comment]
@jfischer-phytec-iot as nobody objects and I succesfully tested this PR I'd give my ACK. Will merge after squash

**Comment by jfischer-phytec-iot** [comment]
@PeterKietzmann squashed

**Comment by PeterKietzmann** [comment]
OK let's do it! &go

**Pull request merged**